docs/custom-capo.md

```diff
@@ -0,0 +1,58 @@
+# Creating a custom cluster api provider ovirt and adding it to the cluster
+
+When we develop the CAPO its usually helpful to test it against a running cluster.
+The following doc explains how to do it.
+
+## Replacing the docker image
+
+1.Lunch a cluster:
+
+Start an Openshift cluster on top of oVirt as usual with the required release image.
+
+2. Create the docker image:
+
+** replace ${username} and ${tag} with the correct values
+```bash
+podman login quay.io
+podman build -t quay.io/${username}/cluster-api-provider-ovirt:${tag} .
+podman push quay.io/${username}/cluster-api-provider-ovirt:${tag}
+```
+
+3. Stop Cluster Version Operator
+In the running cluster cluster-version-operator is responsible for maintaining functioning and non-altered elements.
+In that case to be able to use custom operator image one has to perform one of these operations:
+Set your operator in umanaged state, see here for details, in short:
+```yaml
+oc patch clusterversion/version --type='merge' -p "$(cat <<- EOF
+spec:
+  overrides:
+  - group: apps/v1
+    kind: Deployment
+    name: machine-api-operator
+    namespace: openshift-machine-api
+    unmanaged: true
+EOF
+)"
+```
+
+Scale down cluster-version-operator:
+```bash
+oc scale --replicas=0 deploy/cluster-version-operator -n openshift-cluster-version
+```
+
+IMPORTANT: This apprach disables cluster-version-operator completly, whereas previous only tells it to not manage a kube-scheduler-operator!
+
+4. Replace the CAPO image
+
+```bash
+oc -n openshift-machine-api edit configmap/machine-api-operator-images
+```
+and replace clusterAPIControllerOvirt with the pull (by digest) of your image
```

Comment:
Users and devel will ask: How can I find the digest from the my image? Can we add an example here?